### PR TITLE
Vs/comet

### PR DESCRIPTION
**Pull request by VishwangiShah312**
Made changes to broker.py to pass the Redis host and port as environment variables.

### CODE REVIEW

File: comet/broker.py
Comment (aelanman) [inline comet/broker.py:41]: In normal operation, these environmental variables may not be in the host environment. To ensure a default value, you can replace these calls with `os.environ.get("REDIS_PORT", DEFAULT_PORT)` (and equivalent for host).

**Review by aelanman — COMMENTED** [review]
Nice work, @VishwangiShah312 ! One minor comment but otherwise it looks fine to me.

**Comment by jrs65** [comment]
@VishwangiShah312 as @aelanman says you need to make sure that any changes you make will not effect the current behaviour. This change would have broken the CHIME correlator if deployed as is. 

Also, if you're going to be upstreaming changes (which I very much encourage) I'm going to ask that you get a second review from someone more familiar with the code internals. In practice that is going to mean myself, or if I'm not available @andrerenard, @anjakefala or @ketiltrout . The same is also going to be true of coco.

Again I definitely encourage you to have this stuff upstreamed, but as it really is at the heart of our acquisition system I want to be sure that nothing inadvertently breaks.

**Comment by jrs65** [comment]
We should also get the tests going again. We haven't done much development to comet since Travis shutdown, hence the tests don't run. They would have caught the error in this PR.

File: comet/broker.py
Comment (VishwangiShah312) [reply comet/broker.py:41]: Thanks Adam! Will do that.

**Comment by VishwangiShah312** [comment]
@jrs65 Sorry about that. I will definitely make sure to get a second review before upstreaming changes.

**Comment by aelanman** [comment]
Hi @jrs65 -- Thanks for looking this over. How should we go about verifying that the tests pass?

**Review by andrerenard — CHANGES_REQUESTED** [review]
Can we have a default value if the environment variables aren't set, and/or make sure the scripts for the CHIME setup set the environment variables (e.g. adding something to ch_ansible)?    On a system without the environment setup, I think this will cause comet to fail.

**Comment by andrerenard** [comment]
As for the unit tests, it would be good at the very least to run the tests manually. You should be able to follow what's in the Travis configuration: https://github.com/chime-experiment/comet/blob/master/.travis.yml   Ideally we'd port that to github actions, but I'm too sure who should take on that work.  If you want to give that a try it would be much appreciated.

**Comment by jrs65** [comment]
Yeah, as @andrerenard says, the way to get the tests working is to port the travis.yml to a Github actions workflow. If I feel bored and motivated in exactly the right amounts I may start some of that today.

**Comment by jrs65** [comment]
But you're welcome to give it a go if you want!

**Comment by aelanman** [comment]
@jrs65 @andrerenard 
Some progress has been made on this lately! Vishwangi ported the travis setup to a github action config and I worked through to get the tests passing. It does currently skip a couple tests because those tests require chimedb access, which it doesn't have and I'm not sure how to make that work.

**Comment by aelanman** [comment]
Correction -- I was mistaken. The issue was with the fake chimedb that was being set up in the test environment. I've now fixed those tests as well, so all the unit tests are passing in github actions.

**Comment by ljgray** [comment]
It would be good to finally get this merge in. I just have a couple of quick comments @VishwangiShah312 @aelanman 

- The commit history should be tidied up
- We should confirm that the updated tests are still working, since it's been a few months. The new workflow should run automatically if a rebased commit is pushed here, so it's worth just doing that first

**Comment by aelanman** [comment]
@ljgray Thanks for reminding me of this PR. I just rebased to clean up the commit history, and I'm hoping that the tests all still pass.

**Comment by jrs65** [comment]
@aelanman I was just trying to push a few fixes to get the tests working and cleanup the workflow file, there might be a few changes coming in a minute or two.

**Comment by jrs65** [comment]
Hmmm... there's now some strange multiprocessing + sanic issue. I've got a crashing setup going on my laptop so I'll resume debugging tomorrow.

**Comment by jrs65** [comment]
Closing as I've cleaned this up in #105

**Pull request closed without merging**
